network/src/node.rs

```diff
@@ -137,7 +137,7 @@ impl Node {
             terminator: Arc::new(AtomicBool::new(false)),
         }));
 
-        if node.config.is_crawler() {
+        if node.config.is_of_type(NodeType::Crawler) {
```

Comment:
sorry I haven't noticed this before, but perhaps we can move this method up to `Node`? it would shorten the calls

Reply:
Sure!